new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiChangeAdminCredentials.java

```diff
@@ -0,0 +1,539 @@
+
+  }
+  
+  /**
+   * Patch domain resource with a new WebLogic domain credentials secret and a new restartVersion,
+   * and verify if the domain spec has been correctly updated.
+   * 
+   * @param domainUid name of the domain resource
+   * @param namespace Kubernetes namespace that the domain is hosted
+   * @param adminServerPodName name of the WebLogic admin server
+   * @param managedServerPrefix prefix of the managed servers
+   * @param replicaCount number of managed servers to start
+   * @param secretNmae name of the secret that is used to patch the domain resource
+   * @return restartVersion of the domain resource
+   */
+  private String patchDomainWithNewSecretAndVerify(
+      final String domainUid,
+      final String namespace,
+      final String adminServerPodName,
+      final String managedServerPrefix,
+      final int replicaCount,
+      final String secretName
+  ) {
+    logger.info(
+        "Patch domain resource {0} in namespace {1} to use the new secret {2}",
+        domainUid, namespace, secretName);
+
+    String restartVersion = patchDomainResourceWithNewAdminSecret(domainUid, namespace, secretName);
+    
+    logger.info(
+        "Check that domain resource {0} in namespace {1} has been patched with new secret {2}",
+        domainUid, namespace, secretName);
+    checkDomainCredentialsSecretPatched(domainUid, namespace, secretName);
+
+    // check and wait for the admin server pod to be patched with the new secret
+    logger.info(
+        "Check that admin server pod for domain resource {0} in namespace {1} has been patched with {2}: {3}",
+        domainUid, namespace, "/spec/webLogicCredentialsSecret/name", secretName);
+
+    return restartVersion;
+  }
+
+  /**
+   * Check that domain resource has been updated with the new WebLogic domain credentials secret.
+   * 
+   * @param domainUid name of the domain resource
+   * @param namespace Kubernetes namespace that the domain is hosted
+   * @param newValue new secret name for the WebLogic domain credentials secret
+   */
+  private void checkDomainCredentialsSecretPatched(
+      String domainUid,
+      String namespace,
+      String newValue
+  ) {
+   
+    // check if domain resource has been patched with the new secret
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for domain {0} to be patched in namespace {1} "
+            + "(elapsed time {2}ms, remaining time {3}ms)",
+            domainUid,
+            namespace,
+            condition.getElapsedTimeInMS(),
+            condition.getRemainingTimeInMS()))
+        .until(assertDoesNotThrow(() -> domainResourceCredentialsSecretPatched(domainUid, namespace, newValue),
+            String.format(
+               "Domain %s in namespace %s is not patched with admin credentials secret %s",
+               domainUid, namespace, newValue)));
+
+  }
+  
+  private void checkPodRestarted(
```

Comment:
Do we have shared methods for this now?

Reply:
There is no such method in the common test util yet. Would you like me to move this method there?

Reply:
If what you are asking the underlying method, then yes, it shares the same TestAssertions method isPodRestarted with other test classes, for example, ItMiiConfigMapOverride.

Reply:
I'll move this to CommonTestUtils class.

Reply:
moved